### ISSUE TITLE
Best practice for lazy loading view models

### ISSUE THREAD

**Issue by manzanotti** [finnsson/pagerjs#166]
I've got pager.js working with a big project, and have started having a few issues that have made me question how we've got it set up. What is the best practice for lazy loading view models that need to get data from an AJAX call?

I'm not using require.js, as we prefer the versioning and automatic bundling/minification that Asp.Net gives us, but we create all the view models when the SPA is loaded, and have hooked up an onNavigate method that calls the method on the view model that gets data via an AJAX call.

Obviously, this now seems like a bad idea, as we're creating thousands of variables and functions on loading the app, so I'd like to move to the lazy loading method but the documentation is a little basic, so I'm struggling to translate it to our app.

Basically, should I create the view model when it's requested by withOnShow, and make the construction of the object make the AJAX call? If it's a bad idea to create a new one every time, then how do I make the AJAX call the second time I visit the child page using lazy loading?
